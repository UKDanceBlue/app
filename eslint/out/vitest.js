import eslintPluginVitest from "eslint-plugin-vitest";
const vitestConfig = {
    files: ["**/*.test.ts", "**/*.test.tsx"],
    plugins: {
        vitest: eslintPluginVitest,
    },
    rules: {
        ...eslintPluginVitest.configs.recommended.rules,
        "vitest/consistent-test-filename": "warn",
        "vitest/consistent-test-it": "warn",
        "vitest/max-nested-describe": "warn",
        "vitest/no-alias-methods": "warn",
        "vitest/no-conditional-expect": "warn",
        "vitest/no-conditional-in-test": "warn",
        "vitest/no-conditional-tests": "warn",
        "vitest/no-disabled-tests": "warn",
        "vitest/no-duplicate-hooks": "warn",
        "vitest/no-focused-tests": "warn",
        "vitest/no-interpolation-in-snapshots": "warn",
        "vitest/no-large-snapshots": "warn",
        "vitest/no-mocks-import": "warn",
        "vitest/no-restricted-matchers": "warn",
        "vitest/no-restricted-vi-methods": "warn",
        "vitest/no-standalone-expect": "warn",
        "vitest/no-test-prefixes": "warn",
        "vitest/no-test-return-statement": "warn",
        "vitest/prefer-called-with": "warn",
        "vitest/prefer-comparison-matcher": "warn",
        "vitest/prefer-each": "warn",
        "vitest/prefer-equality-matcher": "warn",
        "vitest/prefer-expect-assertions": [
            "warn",
            {
                onlyFunctionsWithAsyncKeyword: true,
                onlyFunctionsWithExpectInCallback: true,
            },
        ],
        "vitest/prefer-expect-resolves": "warn",
        "vitest/prefer-hooks-in-order": "warn",
        "vitest/prefer-hooks-on-top": "warn",
        "vitest/prefer-lowercase-title": "warn",
        "vitest/prefer-mock-promise-shorthand": "warn",
        "vitest/prefer-snapshot-hint": "warn",
        "vitest/prefer-spy-on": "warn",
        "vitest/prefer-strict-equal": "warn",
        "vitest/prefer-to-be": "warn",
        "vitest/prefer-to-be-object": "warn",
        "vitest/prefer-to-contain": "warn",
        "vitest/prefer-to-have-length": "warn",
        "vitest/prefer-todo": "warn",
        "vitest/require-hook": "warn",
        "vitest/require-to-throw-message": "warn",
        "vitest/require-top-level-describe": "warn",
    },
    languageOptions: {
        globals: eslintPluginVitest.environments.env.globals,
    },
    // TODO: switch the mobile tests over to vitest
    ignores: ["**/mobile/**"],
};
export default vitestConfig;
