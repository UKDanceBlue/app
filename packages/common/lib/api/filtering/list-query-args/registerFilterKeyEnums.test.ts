import { describe, vi } from "vitest";

const registerEnumType = vi.fn();

vi.doMock("type-graphql", () => ({
  registerEnumType,
}));

const { registerFilterKeyEnums } = await import("./registerFilterKeyEnums.js");

describe("registerFilterKeyEnums", (test) => {
  test("registers nothing when called empty", ({ expect }) => {
    registerFilterKeyEnums([], "Empty", [], [], [], [], []);
    expect(registerEnumType.mock.calls.length).toBe(0);
  });

  test("to register the correct names", ({ expect }) => {
    registerFilterKeyEnums(
      ["str", "oneOf", "num", "date", "bool"],
      "Test",
      ["str"],
      ["oneOf"],
      ["num"],
      ["date"],
      ["bool"]
    );
    expect(registerEnumType.mock.calls);
  });

  test("to register the correct names when only some are empty", ({
    expect,
  }) => {
    registerFilterKeyEnums(
      ["str", "oneOf", "num", "date", "bool"],
      "Test",
      [],
      ["oneOf"],
      [],
      ["date"],
      []
    );
    expect(registerEnumType.mock.calls).toMatchSnapshot();
  });
});
