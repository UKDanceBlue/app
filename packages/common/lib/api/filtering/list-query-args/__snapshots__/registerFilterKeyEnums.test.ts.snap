// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`registerFilterKeyEnums > to register the correct names 1`] = `
[
  [
    {
      "bool": "bool",
      "date": "date",
      "num": "num",
      "oneOf": "oneOf",
      "str": "str",
    },
    {
      "name": "TestAllKeys",
    },
  ],
  [
    {
      "str": "str",
    },
    {
      "name": "TestStringFilterKeys",
    },
  ],
  [
    {
      "oneOf": "oneOf",
    },
    {
      "name": "TestOneOfFilterKeys",
    },
  ],
  [
    {
      "num": "num",
    },
    {
      "name": "TestNumericFilterKeys",
    },
  ],
  [
    {
      "date": "date",
    },
    {
      "name": "TestDateFilterKeys",
    },
  ],
  [
    {
      "bool": "bool",
    },
    {
      "name": "TestBooleanFilterKeys",
    },
  ],
]
`;

exports[`registerFilterKeyEnums > to register the correct names when only some are empty 1`] = `
[
  [
    {
      "bool": "bool",
      "date": "date",
      "num": "num",
      "oneOf": "oneOf",
      "str": "str",
    },
    {
      "name": "TestAllKeys",
    },
  ],
  [
    {
      "str": "str",
    },
    {
      "name": "TestStringFilterKeys",
    },
  ],
  [
    {
      "oneOf": "oneOf",
    },
    {
      "name": "TestOneOfFilterKeys",
    },
  ],
  [
    {
      "num": "num",
    },
    {
      "name": "TestNumericFilterKeys",
    },
  ],
  [
    {
      "date": "date",
    },
    {
      "name": "TestDateFilterKeys",
    },
  ],
  [
    {
      "bool": "bool",
    },
    {
      "name": "TestBooleanFilterKeys",
    },
  ],
  [
    {
      "bool": "bool",
      "date": "date",
      "num": "num",
      "oneOf": "oneOf",
      "str": "str",
    },
    {
      "name": "TestAllKeys",
    },
  ],
  [
    {
      "oneOf": "oneOf",
    },
    {
      "name": "TestOneOfFilterKeys",
    },
  ],
  [
    {
      "date": "date",
    },
    {
      "name": "TestDateFilterKeys",
    },
  ],
]
`;
